velox/common/memory/MmapAllocator.cpp

```diff
@@ -384,31 +425,112 @@ bool MmapAllocator::SizeClass::allocateLocked(
     }
     uint64_t bits = pageAllocated_[cursor];
     if (bits != kAllSet) {
-      if (considerMappedOnly > 0) {
-        uint64_t mapped = pageMapped_[cursor];
-        uint64_t mappedFree = ~bits & mapped;
-        if (mappedFree == 0) {
-          continue;
+      int previousToGo = numPagesToGo;
+      assert(numUnmapped != nullptr);
+      allocateAny(cursor, numPagesToGo, *numUnmapped, out);
+      numAllocatedUnmapped_ += previousToGo - numPagesToGo;
+      if (numPagesToGo == 0) {
+        clockHand_ = previousCursor;
```

Comment:
It seems like we did not update clockHand_ in previous version. I wonder this contributes to performance gain or not.